src/otx/core/model/entity/segmentation.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Class definition for detection model entity used in OTX."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.segmentation import SegBatchDataEntity, SegBatchPredEntity
+from otx.core.model.entity.base import OTXModel
+from otx.core.utils.build import build_mm_model
+
+if TYPE_CHECKING:
+    from mmseg.models.data_preprocessor import SegDataPreProcessor
+    from omegaconf import DictConfig
+    from torch import nn
+
+
+class OTXSegmentationModel(OTXModel[SegBatchDataEntity, SegBatchPredEntity]):
+    """Base class for the detection models used in OTX."""
+
+
+class MMSegCompatibleModel(OTXSegmentationModel):
+    """Segmentation model compatible for MMSeg.
+
+    It can consume MMSeg model configuration translated into OTX configuration
+    (please see otx.tools.translate_mmrecipe) and create the OTX detection model
+    compatible for OTX pipelines.
+    """
+
+    def __init__(self, config: DictConfig) -> None:
+        self.config = config
+        self.load_from = self.config.pop("load_from", None)
+        super().__init__()
+
+    def _create_model(self) -> nn.Module:
+        from mmengine.registry import MODELS as MMENGINE_MODELS
+        from mmseg.registry import MODELS
+
+        try:
```

Comment:
Why do we need try/except condition in here?

Reply:
In Unit test this module was registered twice -> error occurred. With next PR with Unit tests I will try to avoid try-except. For this PR, I will remove it